diff --git libdvdnav-4.2.0.orig/src/searching.c libdvdnav-4.2.0/src/searching.c
index 66f7adb..31bf0d3 100644
--- libdvdnav-4.2.0.orig/src/searching.c
+++ libdvdna-4.2.0b/src/searching.c
@@ -628,6 +628,7 @@ uint32_t dvdnav_describe_title_chapters(dvdnav_t *this, int32_t title, uint64_t
     else
       endcellnr = 0;
 
+    if (!pgc->cell_playback) continue;
     do {
       cell = &pgc->cell_playback[cellnr-1];
       if(!(cell->block_type == BLOCK_TYPE_ANGLE_BLOCK &&
